===================================================================
RCS file: /cvs/gnome/gimp/app/core/gimpgradient-load.c,v
retrieving revision 1.11
retrieving revision 1.11.2.1
diff -u -r1.11 -r1.11.2.1
--- gimp/app/core/gimpgradient-load.c	2004/12/16 22:59:27	1.11
+++ gimp/app/core/gimpgradient-load.c	2006/08/05 00:06:51	1.11.2.1
@@ -169,8 +169,9 @@
         }
       else
         {
-	  g_message (_("Corrupt segment %d in gradient file '%s'."),
-		     i, gimp_filename_to_utf8 (filename));
+          g_set_error (error, GIMP_DATA_ERROR, GIMP_DATA_ERROR_READ,
+                       _("Corrupt segment %d in gradient file '%s'."),
+                       i, gimp_filename_to_utf8 (filename));
           g_object_unref (gradient);
           fclose (file);
           return NULL;
